### PR TITLE
[SPARK-30684 ][WEBUI][FollowUp] A new approach for SPARK-30684

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Simplify the changes for adding metrics description for WholeStageCodegen in https://github.com/apache/spark/pull/27405 

### Why are the changes needed?

In https://github.com/apache/spark/pull/27405, the UI changes can be made without using the function `adjustPositionOfOperationName` to adjust the position of operation name and mark as an operation-name class.

I suggest we make simpler changes so that it would be easier for future development.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Manual test with the queries provided in https://github.com/apache/spark/pull/27405 
```
sc.parallelize(1 to 10).toDF.sort("value").filter("value > 1").selectExpr("value * 2").show
sc.parallelize(1 to 10).toDF.sort("value").filter("value > 1").selectExpr("value * 2").write.format("json").mode("overwrite").save("/tmp/test_output")
sc.parallelize(1 to 10).toDF.write.format("json").mode("append").save("/tmp/test_output")
```
image

### CODE REVIEW

File: sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js [hunk 1]
```diff
@@ -82,7 +81,7 @@ function setupTooltipForSparkPlanNode(nodeId) {
  * and sizes of graph elements, e.g. padding, font style, shape.
  */
 function preprocessGraphLayout(g) {
-  g.graph().ranksep = "90";
+  g.graph().ranksep = "70";
```
Comment (gengliangwang) [inline sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js]: setting the value from 90 to 70 to make the graph neater.

**Comment by srowen** [comment]
CC @sarutak

**Comment by sarutak** [comment]
I'll review soon.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala [hunk 2]
```diff
@@ -202,8 +197,8 @@ private[ui] class SparkPlanGraphCluster(
     val labelStr = if (duration.nonEmpty) {
       require(duration.length == 1)
       val id = duration(0).accumulatorId
-      if (metricsValue.contains(id)) {
-        name + "\n \n" + duration(0).name + ": " + metricsValue(id)
+      if (metricsValue.contains(duration(0).accumulatorId)) {
```
Comment (sarutak) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala]: Please keep `metricsBalue.contains(id)` as is. Let's avoid calling `duration(0).accumulatorId`  twice.

File: sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js (hunk 1 above)
Comment (sarutak) [reply sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js]: A gap between an arrow and a node in a whole-stage-codegen node seems to be broader than before (e.g. SerializeFromObject and its forwarding arrow in the screenshot).
It's a little bit ugly so I think it's better to narrow those gaps if you can.

Comment (dongjoon-hyun) [reply sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js]: Got it.

**Review by sarutak — CHANGES_REQUESTED** [review]

**Comment by dongjoon-hyun** [comment]
BTW, @gengliangwang . This is reported as `Improvement` JIRA instead of a follow-up. So, you want this change in `3.1.0` only. Did I understand correctly?

**Comment by gengliangwang** [comment]
> BTW, @gengliangwang . This is reported as Improvement JIRA instead of a follow-up. So, you want this change in 3.1.0 only. Did I understand correctly? We don't backport the improvement.

I see. Let me change it to follow-up.
Thanks for the reminder.

**Comment by gengliangwang** [comment]
@sarutak Yes, the gaps are annoying. 
I have fixed it in the latest commit.

**Review by sarutak — APPROVED** [review]

**Comment by sarutak** [comment]
LGTM.

**Comment by gengliangwang** [comment]
Merging to master/3.0.
Thanks for the review @sarutak @dongjoon-hyun

**Pull request closed without merging**